apps/src/lib/script-editor/FlexGroup.jsx

```diff
@@ -0,0 +1,190 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import _ from 'lodash';
+import color from '../../util/color';
+import { borderRadius, ControlTypes } from './constants';
+import OrderControls from './OrderControls';
+import StageCard from './StageCard';
+import { NEW_LEVEL_ID, addStage, addGroup } from './editorRedux';
+
+const styles = {
+  groupHeader: {
+    fontSize: 18,
+    color: 'white',
+    background: color.cyan,
+    borderTopLeftRadius: borderRadius,
+    borderTopRightRadius: borderRadius,
+    padding: 10
+  },
+  groupBody: {
+    background: color.lightest_cyan,
+    borderBottomLeftRadius: borderRadius,
+    borderBottomRightRadius: borderRadius,
+    padding: 10,
+    marginBottom: 20
+  },
+  addGroup: {
+    fontSize: 14,
+    color: 'white',
+    background: color.cyan,
+    border: `1px solid ${color.cyan}`,
+    boxShadow: 'none',
+    margin: '0 0 30px 0'
+  },
+  addStage: {
+    fontSize: 14,
+    color: '#5b6770',
+    background: 'white',
+    border: '1px solid #ccc',
+    boxShadow: 'none',
+    margin: '0 10px 10px 10px'
+  }
+};
+
+const FlexGroup = React.createClass({
+  propTypes: {
+    addGroup: React.PropTypes.func.isRequired,
+    addStage: React.PropTypes.func.isRequired,
+    stages: React.PropTypes.array.isRequired
+  },
+
+  handleAddGroup() {
+    this.props.addGroup(prompt('Enter new stage name'), prompt('Enter new group name'));
+  },
+
+  handleAddStage(position) {
+    this.props.addStage(position, prompt('Enter new stage name'));
+  },
+
+  serializeStages(stages) {
+    let s = [];
+    stages.forEach(stage => {
+      let t = `stage '${stage.name}'`;
```

Comment:
could we have more descriptive variable names than s and t? (Here and in serializeLevel)